pkg/asset/cluster/azure/azure.go

```diff
@@ -12,3 +21,44 @@ func Metadata(config *types.InstallConfig) *azure.Metadata {
 		Region: config.Platform.Azure.Region,
 	}
 }
+
+// PreTerraform performs any infrastructure initialization which must
+// happen before Terraform creates the remaining infrastructure.
+func PreTerraform(ctx context.Context, clusterID *installconfig.ClusterID, installConfig *installconfig.InstallConfig) error {
+	clusterName := installConfig.Config.ObjectMeta.Name
+	azureDNS, _ := azureconfig.NewDNSConfig()
+	record := fmt.Sprintf("api.%s", clusterName)
+
+	exists := false
+	rgName := installConfig.Config.Azure.BaseDomainResourceGroupName
+	zoneName := installConfig.Config.BaseDomain
+
+	// Look for an existing CNAME first
+	rs, err := azureDNS.GetDNSRecordSet(rgName, zoneName, record, azdns.CNAME)
+	if err == nil && rs.CnameRecord != nil {
+		exists = true
+	}
+
+	// Look for an A record
+	if exists != true {
+		rs, err := azureDNS.GetDNSRecordSet(rgName, zoneName, record, azdns.A)
+		if err == nil && rs.ARecords != nil && len(*rs.ARecords) > 0 {
+			exists = true
+		}
+	}
+
+	// Look for an AAAA record
+	if exists != true {
+		rs, err := azureDNS.GetDNSRecordSet(rgName, zoneName, record, azdns.AAAA)
+		if err == nil && rs.AaaaRecords != nil && len(*rs.AaaaRecords) > 0 {
+			exists = true
+		}
+	}
+
+	// If a record exists, it's likely a cluster already exists
+	if exists == true {
+		return errors.New(fmt.Sprintf("cluster '%s' already exists", clusterName))
```

Comment:
i think the error should be `api.clustername record already exists in <dns zone name or id> and might be in use by another cluster, please remove it to continue`

Reply:
Fixed.